src/main/java/producer/SerializedProducerRecord.java

```diff
@@ -0,0 +1,45 @@
+package producer;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import commons.header.Header;
+import commons.headers.Headers;
+import commons.serializers.OptionalSerializer;
+import commons.serializers.HeaderSerializer;
+import commons.serializers.HeadersSerializer;
+import commons.serializers.ProducerRecordSerializer;
+
+import java.util.Optional;
+
+public class SerializedProducerRecord {
+    private static Kryo kryo;
+
+    static {
+        // All classes will share a single Kryo object.
+        kryo = new Kryo();
+
+        // Kyro requires each class to be registered for serialization. (Includes nested objects)
+        kryo.register(ProducerRecord.class, new ProducerRecordSerializer<>(String.class, Integer.class));
+        kryo.register(Optional.class, new OptionalSerializer());
```

Comment:
OOC, does this mean producer record serialization would only work for <String,Integer> instances?

Reply:
Mainly just wondering what the String.class and Integer.class type params would mean in the context of the serialization

Reply:
No at least `ProducerRecordSerializer` allows for most types. I'm not sure about the other components of the application. But I wanted to make it as generic as possible for future refactor that may go into this. Check out the constructor below:
```java
    private final Class<K> keyClass;
    private final Class<V> valueClass;

    public ProducerRecordSerializer(Class<K> keyClass, Class<V> valueClass) {
        this.keyClass = keyClass;
        this.valueClass = valueClass;
    }
```

Reply:
Got it, so it's not needed?

Reply:
I saw that constructor you posted earlier, which is why I was wondering if that ` kryo.register(ProducerRecord.class, new ProducerRecordSerializer<>(String.class, Integer.class));` line is needed

Reply:
> I saw that constructor you posted earlier, which is why I was wondering if that ` kryo.register(ProducerRecord.class, new ProducerRecordSerializer<>(String.class, Integer.class));` line is needed

for `kryo` its needed because i believe it doesn't support generics as natively since we are extending `Serializer`

Reply:
I see, what would happen if we try to serialize/deserialize a `ProducerRecord<String, String>`?

Reply:
> I see, what would happen if we try to serialize/deserialize a `ProducerRecord<String, String>`?

Uploaded fix on the latest commit, good catch!

Reply:
>Uploaded fix on the latest commit, good catch!

I believe we'll still get other errors but the whole serialization thing can be followed up on in a separate PR to unblock this PR